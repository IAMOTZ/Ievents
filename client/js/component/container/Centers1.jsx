import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
// Actions.
import { getAllCenters, showCenterModal } from '../../actions/centerActions';
// Common components.
import TopNavigation from '../common/TopNavigation.jsx';
import Footer from '../common/Footer.jsx';
import CenterCards from '../common/CenterCards.jsx';
import { CenterModal } from '../common/Modal';
import { LoadingContainer } from '../common/LoadingAnimation.jsx';

@connect(store => (
  {
    centers: store.centers.centers,
    modalContent: store.centers.modalContent ? store.centers.modalContent : {
      name: null,
      images: [],
      details: null,
      capacity: null,
      price: null,
      bookedOn: [],
      type: null,
    },
    isUserFetched: store.user.status.fetched,
    status: {
      fetching: store.centers.status.fetching,
    },
  }
))
export default class Centers1 extends React.Component {
  componentWillMount() {
    this.props.dispatch(getAllCenters());
  }

  /**
   * Displays the center modal.
   * It uses the center ID to get the details of the center to show.
   * @param {Event} event The event object.
   */
  showModal = (event) => {
    this.props.dispatch(showCenterModal(event.target.id));
  }

  render() {
    let component;
    if (this.props.isUserFetched) {
      component = (
        <Redirect to="/addEvent" />
      );
    } else {
      component = (
        <div id="centers1-container">
          <TopNavigation />
          {
            this.props.centers.length === 0 && this.props.status.fetching ?
              <LoadingContainer iconSize={4} /> :
              <section id="centers-section" className="mb-5">
                <div className="container">
                  {/* Search Box */}
                  <div
                    id="center-section-content"
                    className="d-flex flex-column align-items-center"
                  >
                    <div id="center-section-header">
                      <h1 className="display-4">A special center for a special event</h1>
                    </div>
                    <div className="search-box input-group w-50 mt-5">
                      <input
                        type="text"
                        className="form-control"
                        placeholder="Search for a center"
                      />
                      <div className="input-group-btn d-none d-sm-block">
                        <button
                          type="button"
                          className="btn btn-secondary dropdown-toggle"
                          data-toggle="dropdown"
                        >Search with
                        </button>
                        <div className="dropdown-menu dropdown-menu-right">
                          <a className="dropdown-item" href="#">Location</a>
                          <div role="separator" className="dropdown-divider" />
                          <a className="dropdown-item" href="#">Name</a>
                        </div>
                      </div>
                    </div>
                    <div className="search-box mt-3 d-block d-sm-none">
                      <button
                        type="button"
                        className="btn btn-secondary dropdown-toggle"
                        data-toggle="dropdown"
                      >Search with
                      </button>
                      <div className="dropdown-menu">
                        <a className="dropdown-item" href="#">Location</a>
                        <div role="separator" className="dropdown-divider" />
                        <a className="dropdown-item" href="#">Name</a>
                      </div>
                    </div>
                  </div>
                  {/* Centers Grid */}
                  <div className="mt-5">
                    <div className="row">
                      <CenterCards centers={this.props.centers} btnAction={this.showModal} />
                    </div>
                  </div>
                </div>
              </section>
          }
          {/* Modal */}
          <CenterModal
            modalContent={this.props.modalContent}
            redirectPath="/users/login"
          />
          <Footer />
        </div>
      );
    }
    return component;
  }
}
