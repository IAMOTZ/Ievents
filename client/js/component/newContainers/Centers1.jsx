import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { find } from 'lodash';
import { getAllCenters } from '../../actions/centerActions';
import { setCenterToBook } from '../../actions/eventActions';
import TopNavigation from '../common/TopNavigation';
import Footer from '../common/Footer';
import CenterCards from '../common/CenterCards';
import { CenterModal } from '../common/Modal';
import { LoadingContainer } from '../common/LoadingAnimation';

@connect(store => (
  {
    isUserAuthenticaed: store.authReducer.loggingUserResolved,
    fetchingCenterStarted: store.fetchCentersReducer.fetchingCenterStarted,
    centers: store.fetchCentersReducer.centers,
  }
))
class Centers1 extends React.Component {
  constructor() {
    super();
    this.state = {
      modalContent: null,
    };
  }
  componentWillMount() {
    this.props.dispatch(getAllCenters());
  }

  /**
   * It updates the store about a center that is to be booked.
   * @param {Number} centerId The ID of the center.
   */
  onBook = (centerId) => {
    this.props.dispatch(setCenterToBook(centerId));
  }

  /**
   * Displays the center modal.
   * It uses the center ID to get the details of the center to show.
   * @param {Event} event The event object.
   */
  showModal = (event) => {
    const state = { ...this.state };
    const centerId = Number(event.target.id);
    state.modalContent = find(this.props.centers, { id: centerId });
    this.setState(state);
  }

  render() {
    if (this.props.isUserAuthenticaed) {
      return (<Redirect to="/addEvent" />);
    }
    return (
      <div id="centers1-container">
        <TopNavigation />
        {
          this.props.centers.length === 0 && this.props.fetchingCenterStarted ?
            <LoadingContainer iconSize={4} /> :
            <section id="centers-section" className="mb-5">
              <div className="container">
                {/* Search Box */}
                <div
                  id="center-section-content"
                  className="d-flex flex-column align-items-center"
                >
                  <div id="center-section-header">
                    <h1 className="display-4">A special center for a special event</h1>
                  </div>
                  <div className="search-box input-group w-50 mt-5">
                    <input
                      type="text"
                      className="form-control"
                      placeholder="Search for a center"
                    />
                    <div className="input-group-btn d-none d-sm-block">
                      <button
                        type="button"
                        className="btn btn-secondary dropdown-toggle"
                        data-toggle="dropdown"
                      >Search with
                        </button>
                      <div className="dropdown-menu dropdown-menu-right">
                        <a className="dropdown-item" href="#">Location</a>
                        <div role="separator" className="dropdown-divider" />
                        <a className="dropdown-item" href="#">Name</a>
                      </div>
                    </div>
                  </div>
                  <div className="search-box mt-3 d-block d-sm-none">
                    <button
                      type="button"
                      className="btn btn-secondary dropdown-toggle"
                      data-toggle="dropdown"
                    >Search with
                      </button>
                    <div className="dropdown-menu">
                      <a className="dropdown-item" href="#">Location</a>
                      <div role="separator" className="dropdown-divider" />
                      <a className="dropdown-item" href="#">Name</a>
                    </div>
                  </div>
                </div>
                {/* Centers Grid */}
                <div className="mt-5">
                  <div className="row">
                    <CenterCards centers={this.props.centers} btnAction={this.showModal} />
                  </div>
                </div>
              </div>
            </section>
        }
        {/* Modal */}
        {
          this.state.modalContent ? <CenterModal
            modalContent={this.state.modalContent}
            onBook={this.onBook}
            redirectPath="/users/login"
          /> : null
        }
        <Footer />
      </div>
    );
  }
}

export default Centers1;
