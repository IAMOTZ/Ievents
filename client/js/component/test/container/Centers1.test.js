/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Centers1 from '../../container/Centers1.jsx';

const props = {
  centers: [
    { id: 1, name: 'test-center1' },
    { id: 2, name: 'test-center2' },
  ],
  modalContent: {
    name: null,
    images: [],
    details: null,
    capacity: null,
    price: null,
    bookedOn: [],
    type: null,
  },
  isUserFetched: false,
  status: {
    fetching: false,
  },
  dispatch: () => { },
}

const alterProps = newProps => ({
  ...props,
  centers: newProps.centers || props.centers,
  isUserFetched: newProps.isUserFetched || false,
  status: {
    fetching: newProps.fetching || false,
  },
});

describe('<Centers1 />', () => {
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Centers1 {...props} />);
    });
    it('should render well on page load', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to addEvent page if user is already logged in', () => {
      wrapper.setProps(alterProps({ isUserFetched: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render well when fetching centers', () => {
      wrapper.setProps(alterProps({ fetching: true, centers: [] }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
