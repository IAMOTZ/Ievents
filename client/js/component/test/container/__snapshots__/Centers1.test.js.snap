// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Centers1 /> Rendering: should redirect to addEvent page if user is already logged in 1`] = `
<Redirect
  push={false}
  to="/addEvent"
/>
`;

exports[`<Centers1 /> Rendering: should render well on page load 1`] = `
<div
  id="centers1-container"
>
  <TopNav />
  <section
    className="mb-5"
    id="centers-section"
  >
    <div
      className="container"
    >
      <div
        className="d-flex flex-column align-items-center"
        id="center-section-content"
      >
        <div
          id="center-section-header"
        >
          <h1
            className="display-4"
          >
            A special center for a special event
          </h1>
        </div>
        <div
          className="search-box input-group w-50 mt-5"
        >
          <input
            className="form-control"
            placeholder="Search for a center"
            type="text"
          />
          <div
            className="input-group-btn d-none d-sm-block"
          >
            <button
              className="btn btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              type="button"
            >
              Search with
            </button>
            <div
              className="dropdown-menu dropdown-menu-right"
            >
              <a
                className="dropdown-item"
                href="#"
              >
                Location
              </a>
              <div
                className="dropdown-divider"
                role="separator"
              />
              <a
                className="dropdown-item"
                href="#"
              >
                Name
              </a>
            </div>
          </div>
        </div>
        <div
          className="search-box mt-3 d-block d-sm-none"
        >
          <button
            className="btn btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            type="button"
          >
            Search with
          </button>
          <div
            className="dropdown-menu"
          >
            <a
              className="dropdown-item"
              href="#"
            >
              Location
            </a>
            <div
              className="dropdown-divider"
              role="separator"
            />
            <a
              className="dropdown-item"
              href="#"
            >
              Name
            </a>
          </div>
        </div>
      </div>
      <div
        className="mt-5"
      >
        <div
          className="row"
        >
          <CenterCards
            btnAction={[Function]}
            centers={
              Array [
                Object {
                  "id": 1,
                  "name": "test-center1",
                },
                Object {
                  "id": 2,
                  "name": "test-center2",
                },
              ]
            }
          />
        </div>
      </div>
    </div>
  </section>
  <CenterModal
    modalContent={
      Object {
        "bookedOn": Array [],
        "capacity": null,
        "details": null,
        "images": Array [],
        "name": null,
        "price": null,
        "type": null,
      }
    }
    redirectPath="/users/login"
  />
  <Footer />
</div>
`;

exports[`<Centers1 /> Rendering: should render well when fetching centers 1`] = `
<div
  id="centers1-container"
>
  <TopNav />
  <LoadingContainer
    iconSize={4}
  />
  <CenterModal
    modalContent={
      Object {
        "bookedOn": Array [],
        "capacity": null,
        "details": null,
        "images": Array [],
        "name": null,
        "price": null,
        "type": null,
      }
    }
    redirectPath="/users/login"
  />
  <Footer />
</div>
`;
